src/js/jsx/sections/layers/LayerFace.jsx

```diff
@@ -167,7 +167,14 @@ define(function (require, exports, module) {
                 }
             }
 
-            this.getFlux().actions.layers.select(this.props.document, this.props.layer, modifier);
+            // The clicked layer may an have out-of-date document models due to
+            // the aggressive SCU method in LayersPanel.
```

Comment:
I think you basically just mean that `LayerFace` should have an `onClick` handler and `LayersPanel` should execute the action. I guess that makes sense? I'm a little reticent to push more complexity up into `LayersPanel` though...